context/context.go

```diff
@@ -181,15 +192,22 @@ type Response struct {
 // started means the response has sent out.
 func (w *Response) Write(p []byte) (int, error) {
 	w.Started = true
+	if !w.wroteHeader {
+		w.ResponseWriter.WriteHeader(w.Status)
+		//prevent multiple response.WriteHeader calls
+		w.wroteHeader = true
+	}
 	return w.ResponseWriter.Write(p)
 }
 
 // WriteHeader sends an HTTP response header with status code,
 // and sets `started` to true.
 func (w *Response) WriteHeader(code int) {
+	if w.Status > 0 {
+		return
+	}
 	w.Status = code
 	w.Started = true
-	w.ResponseWriter.WriteHeader(code)
```

Comment:
hi @astaxie the http status will be set when Write was called,look here:https://github.com/JessonChan/beego/blob/err_ctrler/context/context.go#L195

farther more,at the http/net package https://golang.org/src/net/http/server.go#L756
and https://golang.org/src/net/http/server.go#L791
the chunkWriter will clone the header if the WriteHeader was called ,so we will lost some header key-value if this line was not removed.
for example 
```
package main

import "net/http"

func main() {
	http.HandleFunc("/hello", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Before", "before")
		w.WriteHeader(200)
		w.Header().Set("After", "after")
		w.Write([]byte("Ok"))
	})
	http.ListenAndServe(":8090", nil)
}
```

![image](https://cloud.githubusercontent.com/assets/707691/13591600/22893118-e525-11e5-97dd-650e9cf24d93.png)